crates/ide-assists/src/handlers/inline_call.rs

```diff
@@ -457,14 +457,29 @@ fn inline(
         }
     }
 
+    let is_async_fn = function.is_async(sema.db);
+    if is_async_fn {
+        cov_mark::hit!(inline_call_async_fn);
+        body = make::async_move_block_expr(body.statements(), body.tail_expr()).clone_for_update();
+
+        // Arguments should be evaluated outside the async block, and then moved into it.
+        if !let_stmts.is_empty() {
+            cov_mark::hit!(inline_call_async_fn_with_let_stmts);
+            body.indent(IndentLevel(1));
+            body = make::block_expr(let_stmts, Some(body.into())).clone_for_update();
+        }
+    } else if let Some(stmt_list) = body.stmt_list() {
+        let_stmts.into_iter().for_each(|stmt| stmt_list.push_front(stmt));
```

Comment:
I know this isn't related to the fix, but can you make this code so that it inserts let statements to `body` all at once? We can use `ted::insert_all()` instead of calling `ted::insert()` (internally called by `push_front()`) for each statement.